### PR TITLE
Principles Documentation

### PR DESCRIPTION
**Pull request by pathunstrom**
This is a rough first draft, would love some input on any of these points, or if anyone thinks I'm missing something key.

* [x] Students and Learners First
* [x] Creativity without Limits
* [x] Fun
* [x] Embracing Radical Change

### CODE REVIEW

File: docs/discussion/values/community-values.rst [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+++++++++++++++++++++
+
+Python for game development is scoffed at by many professionals. This culture of
+contempt is often discouraging for new developers who find a tool they love only
+to listen to how bad it is. In pursuit of radical acceptance, we want to avoid
+contempt culture.
+
+Strong opinions are fine. Every member of the core developers has strong
+opinions on software. Being opinionated is appreciated, but don't share your
+contempt for given software or tools here.
```
Comment (astronouth7303) [inline docs/discussion/values/community-values.rst]: Do we want links to this? https://blog.aurynn.com/2015/12/16-contempt-culture/ is the traditional one, but there's potentially other good ones.

File: docs/discussion/values/technical-values.rst [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+The previous values all demonstrate that occasionally we need to make changes
+to ``ppb`` to help it better align with these goals. That process can be fraught
+for many. We have many developers who care about this code base, and seeing
+their work changed can often be difficult. Beyond that emotional cost, changes
+that break existing code makes old tutorials stale and would break our "No
+Apologies" value. So we work fairly hard to maintain old functionality for as
+long as possible, but will eventually need to be removed.
+
+We've chosen the Northern Hemisphere summer for the time we launch breaking
```
Comment (astronouth7303) [inline docs/discussion/values/technical-values.rst]: Let's drop this in favor of another (future) discussion about our release cycle?

File: docs/discussion/values/technical-values.rst [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+we've refined their points to make them more clear.
+
+Students First
++++++++++++++++
+
+We are an education focused project. In *every* API decision that should be the
+first consideration.
+
+By student, we mean children in a classroom, self taught folks picking up via
+books or the internet, or people following along to a live tutorial.
```
Comment (astronouth7303) [inline docs/discussion/values/technical-values.rst]: Do we want a comment about not leaving experienced people behind?

File: docs/discussion/values/technical-values.rst [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
++++++++++++++
+
+The goal of No Apologies is to minimize students asking questions like "why does
+this work differently here versus there?" Our API should make sense both in
+regards to itself, and also in connection to the larger Python community.
+
+Open Endedness
+++++++++++++++++
+
+I don't actually have a good summary of this.
```
Comment (astronouth7303) [inline docs/discussion/values/technical-values.rst]:
```suggestion
We believe in the creativity of people, students, hobbyists, and professionals. We want to limit that creativity as little as reasonably possible. Yes, ``ppb`` is a 2D sprite-based engine, and that is unlikely to change. But we want to make sure that no genre, no style, no skill level is favored over another.
```

(Something something undertale?)

File: docs/discussion/values/technical-values.rst [hunk 5]
```diff
@@ -0,0 +1,67 @@
+Technical Values
+=================
+
+Technical values describe our processes and design principles. They are
+important to guide the trajectory of ``ppb`` as a software tool with a
+sustainable development lifetime.
```
Comment (astronouth7303) [inline docs/discussion/values/technical-values.rst]: Do we have non-design technical values?

File: docs/discussion/values/technical-values.rst [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+By student, we mean children in a classroom, self taught folks picking up via
+books or the internet, or people following along to a live tutorial.
+
+No Apologies
++++++++++++++
+
+The goal of No Apologies is to minimize students asking questions like "why does
+this work differently here versus there?" Our API should make sense both in
+regards to itself, and also in connection to the larger Python community.
```
Comment (astronouth7303) [inline docs/discussion/values/technical-values.rst]: Something about lacking warts? Principle of least surprise?

File: docs/discussion/values/technical-values.rst (hunk 5 above)
Comment (pathunstrom) [reply docs/discussion/values/technical-values.rst]: The big one I want to get in here at some point is some spin on "Let's try" with discussion about the person doing the work on a feature gets extra leeway in how that piece works (IE, the parts not bound by other principles.)

File: docs/discussion/values/index.rst [hunk 7]
```diff
@@ -1,18 +1,72 @@
 Principles and Values
 =========================
 
-When ``ppb`` started it was given a set of principles that would guide it's
-first few years. Those principles are:
+PursuedPyBear is a principles driven project. From its earliest days, it's been
+guided by a set of ideals that suggest the shame and form of any problem we
```
Comment (astronouth7303) [inline docs/discussion/values/index.rst]:
```suggestion
guided by a set of ideals that suggest the shape and form of any problem we
```

> [PPB has] been guided by a set of ideals that suggest the shape and form of any problem we encounter.

Odd phrasing? Did you mean something like "form of solution to any problem"?

File: docs/discussion/values/index.rst [hunk 8]
```diff
@@ -1,18 +1,72 @@
...
+change to both the code our principles.
+
+We have identified four principles as the root of all of the others:
+
+Students and Learners First
+-----------------------------------
+
+Our first commitment is to the new programmers and game makers who have made
+``ppb`` their tool of choice. Whether introduced to them by a teacher or
+discovered on their own, we care about their experiences primarily. We don't
```
Comment (astronouth7303) [inline docs/discussion/values/index.rst]:
```suggestion
discovered on their own, we care about their experiences primarily. But we don't
```

File: docs/discussion/values/index.rst [hunk 9]
```diff
@@ -1,18 +1,72 @@
...
+long term. Being fun means more than just joyful discovery. We care that
+complexity is neatly hidden until it's necessary to be addressed. We want to
+reduce "warts" in the API, increase overall discoverability, and allow you to
+work playfully.
+
+Embracing Radical Change
+--------------------------------
+
+Over time, the needs of ppb as a project and as a community have changed. It
+started as a solo project to encourage reduce of common patterns in one
```
Comment (astronouth7303) [inline docs/discussion/values/index.rst]:
```suggestion
started as a solo project to encourage reuse of common patterns in one
```

File: docs/discussion/values/index.rst [hunk 10]
```diff
@@ -1,18 +1,72 @@
...
+developers workflows. It's now an education focused community with a team of
```
Comment (astronouth7303) [inline docs/discussion/values/index.rst]:
```suggestion
developer's workflows. It's now an education-focused community with a team of
```

File: docs/discussion/values/index.rst (hunk 9 above)
Comment (pathunstrom) [reply docs/discussion/values/index.rst]: Oh, that was two thoughts blending at once. (Reducing common boilerplate/Reusing best practices or something to that effect. It's viable with your suggestion.

Comment (astronouth7303) [reply docs/discussion/values/index.rst]: Whatever you think is best.

**Comment by pathunstrom** [comment]
Leaving a note here: I'm changing Accepting Radical Change to Radical Acceptance, and containing Accepting Change under it.

File: docs/discussion/values/students-first.rst [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+have of how a system works is thrown a curve ball that requires reassessing what
+you know. There are popular wat talks for both Python and Javascript to get a
+feel for what we mean.
+
+One way this bears out is that no matter what level your knowledge of ``ppb``,
+learning something new should only add to that knowledge, not require
+reassessment.
+
+We also try to reduce the number of times a user is forced to ask "why is it
+like this and not like that?"
```
Comment (astronouth7303) [inline docs/discussion/values/students-first.rst]: This just feels like it needs expansion.

File: docs/discussion/values/creativity-without-limits.rst [hunk 12]
```diff
@@ -0,0 +1,56 @@
+Creativity Without Limits
+==============================
+
+Our focus on creativity without limits is about supporting users at all
+skill levels, and to help guide them from their first lines of code through
+contributing to open source.
```
Comment (astronouth7303) [inline docs/discussion/values/creativity-without-limits.rst:6]: Is this really the end goal?

File: docs/discussion/values/creativity-without-limits.rst [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+The primary example of this is in the basic setup for ppb as a simple event loop
+with the update pattern at its core. This is because it's the most generally
+applicable pattern we have available. We provide a multi-phase update system in
+features for games that need the ability to stage updates instead of
+immediately shifting the state.
+
+Support All Users
+----------------------
+
+We're Students First, but students aren't students forever, and we want
```
Comment (astronouth7303) [inline docs/discussion/values/creativity-without-limits.rst:43]: A+

File: docs/discussion/values/creativity-without-limits.rst [hunk 14]
```diff
@@ -0,0 +1,56 @@
...
+``ppb`` to grow with them. From their first tutorials through to their first
+shipped video game, and hopefully: to their first open source contribution.
+This is about making sure the resources and community are there to help
+develop ``ppb`` users.
+
+From the user perspective, clear tutorials and example code. Open
+discussions about the design of the system. Type hinting to allow the tools to
```
Comment (astronouth7303) [inline docs/discussion/values/creativity-without-limits.rst:50]: Ehhh...... type hinting has proven to be extremely difficult for us.

File: docs/discussion/values/creativity-without-limits.rst (hunk 12 above)
Comment (pathunstrom) [reply docs/discussion/values/creativity-without-limits.rst:6]: It is explicitly part of the pitch I've been using in discussions with people.

File: docs/discussion/values/creativity-without-limits.rst (hunk 14 above)
Comment (pathunstrom) [reply docs/discussion/values/creativity-without-limits.rst:50]: Still a thing we put effort into.

**Pull request merged**